src/mam/mod_mam_muc.erl

```diff
@@ -186,19 +186,19 @@ filter_room_packet(Packet, HostType, EventData = #{}) ->
     case IsArchivable of
         true ->
             #{from_nick := FromNick, from_jid := FromJID, room_jid := RoomJID,
-              role := Role, affiliation := Affiliation} = EventData,
+              role := Role, affiliation := Affiliation, timestamp := TS} = EventData,
             archive_room_packet(HostType, Packet, FromNick, FromJID,
-                                RoomJID, Role, Affiliation);
+                                RoomJID, Role, Affiliation, TS);
         false -> Packet
     end.
 
 %% @doc Archive without validation.
 -spec archive_room_packet(HostType :: host_type(),
                           Packet :: packet(), FromNick :: jid:user(),
                           FromJID :: jid:jid(), RoomJID :: jid:jid(),
-                          Role :: mod_muc:role(), Affiliation :: mod_muc:affiliation()) -> packet().
+                          Role :: mod_muc:role(), Affiliation :: mod_muc:affiliation(), integer()) -> packet().
```

Comment:
```suggestion
                          Role :: mod_muc:role(), Affiliation :: mod_muc:affiliation(), pos_integer()) -> packet().
```

Reply:
I think it could be just `integer` because that's how the timestamp is defined in Erlang docs. However, I would put a variable name there, at least `TS :: integer()` for consistency.